setup.py

```diff
@@ -22,6 +22,22 @@
             "ccx = lms.djangoapps.ccx.plugins:CcxCourseViewType",
             "edxnotes = lms.djangoapps.edxnotes.plugins:EdxNotesCourseViewType",
             "instructor = lms.djangoapps.instructor.views.instructor_dashboard:InstructorDashboardViewType",
+            "courseware = openedx.core.djangoapps.content.course_views.tabs:CoursewareTab",
```

Comment:
This doesn't seem correct. You are registering classes here that are not subclasses of ``CourseViewType``. Let's talk this part through and work out how to handle all of these types.